src/components/InputBox.tsx

```diff
@@ -0,0 +1,40 @@
+import { Box, TextField } from "@mui/material";
+import React, { RefObject } from "react";
+
+type InputBoxProps = {
+};
+
+class InputBox extends React.Component<InputBoxProps> {
```

Comment:
see comments on top. If you use TextField as a controlled component, then this file simplifies quite a lot (and you could actually delete it).

Reply:
done; nice - I understand a bit more about controlled components now